### PR TITLE
Added variant to ensure Arm Forge license acceptance

### PR DESCRIPTION
**Pull request by kent-cheung-arm**

### CODE REVIEW

File: var/spack/repos/builtin/packages/arm-forge/package.py [hunk 1]
```diff
@@ -130,6 +131,12 @@ class ArmForge(Package):
         description='Detect available PMU counters via "forge-probe" during install',
     )
 
+    variant(
+        "accept-eula",
+        default=False,
+        description='Accept the EULA',
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/arm-forge/package.py]: Black style check doesn't like single quotes:

```suggestion
        description="Accept the EULA",
```

File: var/spack/repos/builtin/packages/arm-forge/package.py [hunk 2]
```diff
@@ -143,16 +150,30 @@ class ArmForge(Package):
...
         return "https://content.allinea.com/downloads/arm-forge-%s-linux-%s.tar" % (
             version,
             platform.machine(),
         )
 
+    @run_before("install")
+    def abort_without_eula_acceptance(self):
+        install_example = "spack install arm-forge +accept-eula"
+        license_terms_path = os.path.join(self.stage.source_path, "license_terms")
+        if not self.spec.variants['accept-eula'].value:
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/arm-forge/package.py]:
```suggestion
        if not self.spec.variants["accept-eula"].value:
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/arm-forge/package.py [hunk 3]
```diff
@@ -143,16 +150,30 @@ class ArmForge(Package):
         "ALLINEA_LICENSE_FILE",
         "ALLINEA_LICENCE_FILE",
     ]
-    license_url = "https://developer.arm.com/tools-and-software/server-and-hpc/help/help-and-tutorials/system-administration/licensing/arm-licence-server"
+    license_url = "https://developer.arm.com/documentation/101169/latest/Use-Arm-Licence-Server"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/arm-forge/package.py:153]: Confirmed URL exists

**Review by NickRF — DISMISSED** [review]

**Review by tldahlgren — DISMISSED** [review]
LGTM

**Comment by tldahlgren** [comment]
@spackbot run pipeline

**Comment by kent-cheung-arm** [comment]
Thanks, @tldahlgren. Could I trouble you for another review approval please?

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**